akka-docs/rst/scala/code/docs/persistence/PersistenceEventAdapterDocSpec.scala

```diff
@@ -0,0 +1,195 @@
+          case c => persist(c) { e => p.ref ! e }
+        }
+      })
+
+      val p1 = system.actorOf(props)
+      val m1 = Person("Caplin", 42)
+      val m2 = Box(13)
+      p1 ! m1
+      p1 ! m2
+      p.expectMsg(m1)
+      p.expectMsg(m2)
+
+      val p2 = system.actorOf(props)
+      p.expectMsg(m1)
+      p.expectMsg(m2)
+    }
+  }
+
+  "MyManualJsonEventAdapter" must {
+    "demonstrate how to implement a JSON adapter" in {
+      val p = TestProbe()
+
+      val props = Props(new PersistentActor {
+        override def persistenceId: String = "json-actor"
+        override def journalPluginId: String = "akka.persistence.journal.manual-json-store"
+
+        override def receiveRecover: Receive = {
+          case RecoveryCompleted => // ignore...
+          case e                 => p.ref ! e
+        }
+
+        override def receiveCommand: Receive = {
+          case c => persist(c) { e => p.ref ! e }
+        }
+      })
+
+      val p1 = system.actorOf(props)
+      val m1 = Person("Caplin", 42)
+      val m2 = Box(13)
+      p1 ! m1
+      p1 ! m2
+      p.expectMsg(m1)
+      p.expectMsg(m2)
+
+      val p2 = system.actorOf(props)
+      p.expectMsg(m1)
+      p.expectMsg(m2)
+    }
+  }
+}
+
+trait DomainEvent
+case class Person(name: String, age: Int) extends DomainEvent
+case class Box(length: Int) extends DomainEvent
+
+case class MyTaggingJournalModel(payload: Any, tags: immutable.Set[String])
+
+/**
+ * This is an example adapter which completely takes care of domain<->json translation.
+ * It allows the journal to take care of the manifest handling, which is the FQN of the serialized class.
+ */
+class MyAutoJsonEventAdapter(system: ExtendedActorSystem) extends EventAdapter[JsonElement] {
+
+  val gson = new Gson // TODO: serisouly, simplest to plug in, all type-class based ones fail in this scenario
+
+  override val includeManifest = true
+  override def manifest(event: Any): String =
+    event.getClass.getCanonicalName
+
+  override def toJournal(event: Any): JsonElement =
+    gson.toJsonTree(event)
+
+  override def fromJournal(event: JsonElement, manifest: String): Any = {
+    val clazz = system.dynamicAccess.getClassFor[Any](manifest).get
+    gson.fromJson(event, clazz)
+  }
+
+}
+
+/**
+ * This is an example adapter which completely takes care of domain<->json translation.
+ * It manually takes care of smuggling through the manifest even if the journal does not do anything in this regard,
+ * which is the case in this example since we're persisting to the inmem journal, which does nothing in terms of manifest.
+ */
+class MyManualJsonEventAdapter(system: ExtendedActorSystem) extends EventAdapter[JsonElement] {
+
+  val gson = new Gson // TODO: serisouly, simplest to plug in, all type-class based ones fail in this scenario
+
+  override val includeManifest = false // we take care of it manually
+  override def manifest(event: Any): String = event.getClass.getCanonicalName
```

Comment:
why not always include the manifest, i.e. remove the `def includeManifest`? if not needed it can be `""`